stdlib/Random/src/RNGs.jl

```diff
@@ -316,12 +316,26 @@ function hash_seed(seed::Union{AbstractArray{UInt32}, AbstractArray{UInt64}})
     SHA.digest!(ctx)
 end
 
+function hash_seed(str::AbstractString)
+    ctx = SHA.SHA2_256_CTX()
+    # convert to String such that `codeunits(str)` below is consistent between equal
+    # strings of different types
+    str = String(str)
+    SHA.update!(ctx, codeunits(str))
+    # signature for strings: so far, all hash_seed functions end-up hashing a multiple
+    # of 4 bytes of data, and add the signature (1 byte) at the end; so hash as many
+    # 0x05 bytes as necessary to have a total number of hashed bytes equal to 1 mod 4
+    SHA.update!(ctx, ntuple(Returns(0x05), 5 - mod1(ncodeunits(str), 4)))
```

Comment:
I think this padding with 0x05 will cause collisions between for example "\x05" and "\x05\x05"

Here is one possible solution, where the padding starts with 0x05, then is the number of other extra bytes that were added.

```suggestion
    a = 4 - mod(ncodeunits(str), 4)
    SHA.update!(ctx, (0x05, ntuple(Returns(UInt8(a)),  a)...))
```

Reply:
Actually, is any extra padding needed?

```suggestion
    SHA.update!(ctx, (0x05,))
```

Reply:
Oh right,  i think you are right about the collision! But padding is necessary essentially to avoid collisions with positive integers, which hash a multiple of 4 bytes. So here we want to hash e.g. 4k+1 bytes. To avoid collisions between two strings, we can simply hash also the length of the string.

Reply:
I've committed your suggestion, which avoids having to hash the whole length (`ncodeunits(str)`), and updated to make it collision resistant with integers, by putting 0x05 at the end, and used a loop instead of `ntuple` (type unstable) to make it faster.